### PR TITLE
Humble release versions 2023-11-13

### PR DESCRIPTION
**Pull request by audrow**
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19812/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14314/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20535/)
* RHEL [Build Status](http://ci.ros2.org/job/ci_linux-rhel/514/)

https://ci.ros2.org/view/packaging/job/packaging_linux/3234/
https://ci.ros2.org/view/packaging/job/packaging_linux-aarch64/2598/
https://ci.ros2.org/view/packaging/job/packaging_linux-rhel/1735/
https://ci.ros2.org/view/packaging/job/packaging_windows/3055/
https://ci.ros2.org/view/packaging/job/packaging_windows_debug/2014/

### CODE REVIEW

File: ros2.repos [hunk 1]
```diff
@@ -42,7 +42,7 @@ repositories:
   eclipse-cyclonedds/cyclonedds:
     type: git
     url: https://github.com/eclipse-cyclonedds/cyclonedds.git
-    version: 0.10.3
+    version: 0.10.4
```
Comment (clalancette) [inline ros2.repos:45]: This one we shouldn't update unless we are also going to do a binary release of it.  For now, I think we should skip this and just keep it at 0.10.3.

File: ros2.repos [hunk 2]
```diff
@@ -58,7 +58,7 @@ repositories:
   osrf/osrf_pycommon:
     type: git
     url: https://github.com/osrf/osrf_pycommon.git
-    version: 2.0.2
+    version: 2.1.4
```
Comment (clalancette) [inline ros2.repos:61]: This one we should leave at 2.0.2 (it's one of the few that we don't have a dedicated "humble" branch for).

**Review by clalancette — CHANGES_REQUESTED** [review]
Besides the changes I pointed out inline, we should also probably do a `tinyxml2_vendor` release.

File: ros2.repos (hunk 2 above)
Comment (audrow) [reply ros2.repos]: Fixed in a0b2355eb62288d47aacc0b85b4e6e704d8b6fb3.

File: ros2.repos (hunk 1 above)
Comment (audrow) [reply ros2.repos]: Fixed in a0b2355eb62288d47aacc0b85b4e6e704d8b6fb3.

**Review by clalancette — APPROVED** [review]
This looks good to me pending resolution of the license files in CI.  I'd say let's hold off on merging this for a couple of days to see if we can resolve that.  In the meantime I'll suggest merging in the rosdistro updates so the buildfarm can start rebuilding everything.

**Comment by clalancette** [comment]
So there are still a lot of failing tests on all platforms here.  I think it is worthwhile to re-run the tests from the previous ros2.repos, just to see what the differences are.

**Comment by clalancette** [comment]
I think the problems still have to do with the license for RTI, even though I don't understand why.

For now, I'd say let's go ahead with the patch release even with those, and we can investigate later on.

**Comment by audrow** [comment]
Here is CI with the `humble-release` branch `ros2.repos` file:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19827/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14327/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20548/)

**Comment by clalancette** [comment]
I just pushed 0c456e33e72ed4984d486593635385ad85e51231 to this branch, which updates system_tests to the latest release.  I *believe* that will fix at least some of the warnings we are getting here.  Here is more CI to verify:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19833/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14331/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20555/)

**Pull request merged**